packages/core/src/components/way-input/way-input.tsx

```diff
@@ -0,0 +1,134 @@
+import { Component, Host, h, Prop, Element, Event, EventEmitter, State } from '@stencil/core';
+import { inheritAttributes } from '../../utils/utils';
+@Component({
+  tag: 'way-input',
+  styleUrl: 'way-input.scss',
+  shadow: true,
+})
+export class Wayinput {
+  private inheritedAttributes: { [k: string]: any } = {};
+
+  @Element() el!: HTMLElement;
+
+  /**
+   * The state of the input's value attribute.
+   */
+  @State() value: string;
+
+  /**
+   * Specifies what type of input to use.
+   */
+  @Prop() type: string | undefined;
+
+  /**
+   * The input's name attribute.
+   */
+  @Prop({ reflect: true }) name: string | undefined;
+
+  /**
+   * If `true`, the user cannot interact with the input.
+   */
+  @Prop({ reflect: true }) disabled = false;
+
+  /**
+   * Specifies what if label and input must be inline.
+   */
+  @Prop() inline: boolean;
+
+  /**
+   * The input's size.
+   */
+  @Prop({ reflect: true }) size: 'small' | 'medium' | 'large' = 'medium';
+
+  /**
+   * The input's label. Alternatively, you can use the label slot.
+   */
+  @Prop() label: string | undefined;
+
+
+  /**
+   * Emitted when the input has focus.
+   */
+  @Event() wayFocus!: EventEmitter<void>;
+
+  /**
+   * Emitted when the input loses focus.
+   */
+  @Event() wayBlur!: EventEmitter<void>;
+
+  componentWillLoad() {
+    this.inheritedAttributes = inheritAttributes(this.el, ['aria-label']);
+  }
+
+  private onFocus = () => {
+    this.wayFocus.emit();
+  }
+
+  private onBlur = () => {
+    this.wayBlur.emit();
+  }
+
+  handleChange(event) {
+    this.value = event.target.value;
+
+    // let error = this.error;
+    // if (event.target.validity.typeMismatch) {
+    //   error = true;
+    // }
+  }
+
+  render() {
```

Comment:
**issue**: Hier moet nog code komen om de value als hidden form input buiten de shadow dom te renderen (zodat het te gebruiken is met een reguliere html form submit). Zie mijn WAYDES-17 branch.

Reply:
Is nu toegevoegd, top. De value wordt alleen nooit aangepast door tekst in te typen. Ik zal dit even fixen.